### PR TITLE
[READY] Update README and codecov.yml

### LINKED ISSUE #1499: Minor issue: Possibly outdated readme section regarding clangd support
On [line 101](https://github.com/ycm-core/ycmd/blame/master/README.md#L101) it's written:

    clangd support is currently experimental and changes in the near future might break backwards compatibility.

By reading the readme of the other youcompleteme repo I had the impression clangd was not experimental.

Is it still experimental? If not, then I'd suggest updating the file.

Thanks! I'm new to YCM and I'm figuring stuff out ;)

### PR DESCRIPTION
**Pull request by bstaletic**
Fixes #1499

Just a minor makeover. The codecov.yml change should explain to codecov not to create comments before it receives the entire data.

---
This change is [Reviewable](https://reviewable.io/reviews/ycm-core/ycmd/1516)

### CODE REVIEW

**Review by puremourning — APPROVED** [review]
:lgtm_strong: 

> Reviewed 1 of 2 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/ycm-core/ycmd/1516#-:-MOmgYyt11P6-qXGpRWL:b-uwkowd)* status: 1 of 2 LGTMs obtained (waiting on @bstaletic)

**Comment by bstaletic** [comment]
:lgtm:

**Review by puremourning — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/ycm-core/ycmd/1516#-:-MOmwQNR0OwueBuF4pyB:bnfp4nl)* status: :shipit: complete! 2 of 2 LGTMs obtained (waiting on @bstaletic)

File: README.md [hunk 1]
```diff
@@ -95,15 +95,15 @@ provided in the completion request, other files of the same filetype that were
...
-completion for C-family languages. [clangd][clangd] support is currently
-**experimental** and changes in the near future might break backwards
-compatibility. There's also a Jedi-based completer for semantic completion for
-Python, an OmniSharp-based completer for C#, a [gopls][gopls]-based completer
-for Go (using [gopls][gopls] for jumping to definitions), a TSServer-based
-completer for JavaScript and TypeScript, a [jdt.ls][jdtls]-based server for
-Java, and a [RLS][]-based completer for Rust. More will be added with time.
+There are also several semantic engines in YCM. There's [clangd][clangd]-based
+completer that both provide semantic completion for C-family languages.
+[clangd][clangd] support is currently **experimental** and changes in the near
```
[the remark this answers is not in the document]
Comment (bstaletic) [reply README.md:100]: It is and I somehow forgot why I wanted to make this PR. Anyway, the "experimental bit" was also removed in another commit:

https://github.com/ycm-core/ycmd/commit/2ae3dd022198b12709fd7703a2bc95b9393327bd

**Review by anakimluke — COMMENTED** [review]
I've never used this *review* feature before. Where do I click to close it? 😅

**Comment by bstaletic** [comment]
Github reviews are confusing. I've just clicked on "resolve conversation" below the reply textbox. That's why (among other reasons) we use Reviewable - you would have been able to choose "retract".

**Pull request merged**